frontend/app/component/form/polling_station/PollingStationForm.test.tsx

```diff
@@ -34,32 +34,122 @@ async function fillForm(user: UserEvent, testPollingStation: PollingStation | Om
-      street: "test",
-      postal_code: "1234",
-      locality: "test",
-      polling_station_type: "FixedLocation",
-      number_of_voters: 1,
-      house_number: "test",
-    };
-
-    const onSaved = vi.fn();
-    render(<PollingStationForm electionId={1} onSaved={onSaved} />);
-
-    const user = userEvent.setup();
-    await fillForm(user, testPollingStation);
-
-    await userEvent.click(screen.getByRole("button", { name: "Opslaan en toevoegen" }));
+describe("PollingStationForm", () => {
+  describe("PollingStationForm create", () => {
+    test("Succesful create", async () => {
+      const testPollingStation: Omit<PollingStation, "id"> = {
+        election_id: 1,
+        number: 1,
+        name: "test",
+        street: "test",
+        postal_code: "1234",
+        locality: "test",
+        polling_station_type: "FixedLocation",
+        number_of_voters: 1,
+        house_number: "test",
+      };
+
+      const onSaved = vi.fn();
+      render(<PollingStationForm electionId={1} onSaved={onSaved} />);
+
+      const user = userEvent.setup();
+      await fillForm(user, testPollingStation);
+
+      await userEvent.click(screen.getByRole("button", { name: "Opslaan en toevoegen" }));
+
+      await waitFor(() => {
+        expect(onSaved).toHaveBeenCalledWith(expect.objectContaining(testPollingStation));
+      });
+    });
 
-    await waitFor(() => {
-      expect(onSaved).toHaveBeenCalledWith(expect.objectContaining(testPollingStation));
+    test("Validation", async () => {
+      const testObj: Omit<PollingStation, "id"> = {
+        election_id: 1,
+        number: 42,
+        name: "test",
+        street: "test",
+        postal_code: "1234",
+        locality: "test",
+        polling_station_type: "FixedLocation",
+        number_of_voters: 1,
+        house_number: "test",
+      };
+
+      const onSaved = vi.fn();
+      render(<PollingStationForm electionId={1} onSaved={onSaved} />);
+
+      const user = userEvent.setup();
+      await user.click(screen.getByRole("button", { name: "Opslaan en toevoegen" }));
+
+      const numberInput = screen.getByRole("textbox", { name: "Nummer" });
+      const nameInput = screen.getByRole("textbox", { name: "Naam" });
+      const numberOfVotersInput = screen.getByRole("textbox", { name: "Aantal kiesgerechtigden Optioneel" });
+      const streetInput = screen.getByRole("textbox", { name: "Straatnaam" });
+      const houseNumberInput = screen.getByRole("textbox", { name: "Huisnummer" });
+      const houseNumberAdditionInput = screen.getByRole("textbox", { name: "Toevoeging" });
+      const postalCodeInput = screen.getByRole("textbox", { name: "Postcode" });
+      const localityInput = screen.getByRole("textbox", { name: "Plaats" });
+      const typeInputOption = screen.getByRole("radio", { name: "Vaste locatie" });
+
+      await user.click(typeInputOption);
+
+      await waitFor(() => {
+        expect(numberInput).toHaveAttribute("aria-invalid", "true");
```

Comment:
There's [`toBeInvalid()`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobeinvalid) for this.